src/playercards/AllCardsBagApi.ttslua

```diff
@@ -58,17 +58,69 @@ do
   --     Otherwise, a list of tables, each with the following fields
   --       cardData: TTS object data, suitable for spawning the card
   --       cardMetadata: Table of parsed metadata
-  AllCardsBagApi.getCardsByClassAndLevel = function(class, upgraded)
-    return getAllCardsBag().call("getCardsByClassAndLevel", {class = class, upgraded = upgraded})
+  AllCardsBagApi.getCardsByClass = function(class, includeOthers)
+    return getAllCardsBag().call("getCardsByClass", {class = class, includeOthers = includeOthers})
   end
 
-  AllCardsBagApi.getCardsByCycle = function(cycle)
-    return getAllCardsBag().call("getCardsByCycle", cycle)
+  AllCardsBagApi.getCardsByCycle = function(cycleName, includeOthers)
+    return getAllCardsBag().call("getCardsByCycle", {cycleName = cycleName, includeOthers = includeOthers})
   end
 
   AllCardsBagApi.getUniqueWeaknesses = function()
     return getAllCardsBag().call("getUniqueWeaknesses")
   end
 
+  AllCardsBagApi.getCardsByTrait = function(trait, includeOthers)
+    return getAllCardsBag().call("getCardsByTrait", {trait = trait, includeOthers = includeOthers})
+  end
+
+  AllCardsBagApi.getCardsByIcon = function(icon, count, includeOthers)
+    return getAllCardsBag().call("getCardsByIcon", {icon = icon, count = count, includeOthers = includeOthers})
+  end
+
+  AllCardsBagApi.getCardsByType = function(cardType, includeOthers)
+    return getAllCardsBag().call("getCardsByType", {cardType = cardType, includeOthers = includeOthers})
+  end
+
+  AllCardsBagApi.getCardsByCost = function(cost, includeOthers)
+    return getAllCardsBag().call("getCardsByCost", {cost = cost, includeOthers = includeOthers})
+  end
+
+  AllCardsBagApi.getCardsByLevel = function(level, includeOthers)
+    return getAllCardsBag().call("getCardsByLevel", {level = level, includeOthers = includeOthers})
+  end
+  
+  AllCardsBagApi.getCardsByUses = function(index, includeOthers)
+    return getAllCardsBag().call("getCardsByUses", {index = index, includeOthers = includeOthers})
+  end
+
+  AllCardsBagApi.getPermanents = function(includeOthers)
+    return getAllCardsBag().call("getPermanents", includeOthers)
+  end
```

Comment:
Combine these into a single function that takes the requested index as parameter